engine/src/flutter/shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -344,6 +344,29 @@ private FlutterEngineGroup.Options addEntrypointOptions(FlutterEngineGroup.Optio
     isFlutterEngineFromHost = false;
   }
 
+  // Engine flags can no longer be set via Intent, so warn developers that
+  // engine shell arguments set that way will be ignored.
+  private void warnIfEngineFlagsSetViaIntent(Intent intent) {
+    if (intent.getExtras() == null) {
+      return;
+    }
+
+    Bundle extras = intent.getExtras();
+    Set<String> extrasKeys = extras.keySet();
+
+    for (String extrasKey : extrasKeys) {
+      FlutterShellArgs.Flag flag = FlutterShellArgs.getFlagFromIntentKey(extrasKey);
+      if (flag != null) {
+        Log.w(
+            TAG,
+            "Engine flags can no longer be set via Intent on Android. If you wish to set "
+                + flag.commandLineArgument
+                + ", see flutter/docs/engine/Android-Flutter-Shell-Arguments.md for alternative methods.");
```

Comment:
I assume most use cases for using engine flags are internal, so pointing to an internal doc isn't a huge deal for now, but I'm considering making this a website page too for external developers. Reviewers, let me know if you have thoughts on this.

Reply:
Instead of linking to a path what if you linked to the path on github?